OnlineStore/store/src/main/java/StoreApp.java

```diff
@@ -1,26 +1,58 @@
-import java.lang.reflect.Field;
-import java.util.ArrayList;
+import product.ChainComparator;
+import product.ListProductComparators;
+import product.Product;
+import utility.DOMExecuter;
+
+
+import java.net.URL;
+import java.util.*;
+import java.util.concurrent.*;
 
 public class StoreApp {
     public static void main(String[] args) {
+        /**/
+
+        Store store = new Store(3000);
         RandomStorePopulator populator = new RandomStorePopulator();
-        ArrayList<Product> products = populator.populateStore();
+        List<Product> products = populator.populateStore(90);
         for (int i = 0; i < products.size(); i++) {
             System.out.println(products.get(i));
         }
+        store.setProducts(products);
+
+        URL resource = StoreApp.class.getResource("sort.xml");
+        DOMExecuter executer = new DOMExecuter();
+        Map<String, String> sortCondition = executer.XMLreader(resource.getPath());
 
-        System.out.println("\n sortByPriceAscending ********************************\n");
-        Store store = new Store(products);
-        ArrayList<Product> sortingByPrice = store.sortByPriceAscending();
-        for (int i = 0; i < sortingByPrice.size(); i++) {
-            System.out.println(sortingByPrice.get(i));
+        List<Comparator<Product>> comparatorList = new ListProductComparators(sortCondition).getComparatorList(sortCondition);
+        products.sort(new ChainComparator(comparatorList));
+        System.out.println("\n sorting ********************************\n");
+        for (int i = 0; i < 5; i++) {
+            System.out.println(products.get(i));
         }
 
-        System.out.println("\n sortingByPriceDesc ********************************\n");
-        ArrayList<Product> sortingByPriceDesc = store.sortByPriceDescending();
-        for (int i = 0; i < sortingByPriceDesc.size(); i++) {
-            System.out.println(sortingByPriceDesc.get(i));
+        List<Product> purcheses = new ArrayList<>();
+        System.out.println("purchases:");
+        for (int i = 1; i < 4; i++){// 3 is quantity of consumers
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            //5 is quantity of products, which buy one Consumer
+            int randomInt = (int) (Math.random() * 30);
```

Comment:
and also the same note as for sorter. this code should not be in main class, you should more it some where, think about better place.